tests/helics/application_api/ValueFederateAdditionalTests.cpp

```diff
@@ -414,6 +414,59 @@ BOOST_DATA_TEST_CASE (test_async_calls, bdata::make (core_types), core_type)
     vFed2->finalize ();
 }
 
+
+/** test info field for multiple publications */
+BOOST_DATA_TEST_CASE (test_info_field, bdata::make(core_types_single), core_type) {
+    SetupTest<helics::ValueFederate>(core_type, 1);
+    auto vFed1 = GetFederateAs<helics::ValueFederate>(0);
+
+    auto pubid1 = vFed1->registerPublication<std::string>("pub1");
+    auto pubid2 = vFed1->registerGlobalPublication<int>("pub2");
+    pubid1.setInfo(std::string("test1"));
+    pubid2.setInfo(std::string("test2"));
+    vFed1->enterExecutingMode();
+
+    BOOST_CHECK (vFed1->getCurrentState() == helics::Federate::states::execution);
+
+    auto info1 = vFed1->getInfo(pubid1.getHandle());
+    auto info2 = vFed1->getInfo(pubid2.getHandle());
+    BOOST_CHECK_EQUAL (info1, "test1");
+    BOOST_CHECK_EQUAL (info2, "test2");
+
+    vFed1->finalize();
+}
+
+/** test the pub/sub info field*/
+BOOST_DATA_TEST_CASE (test_info_pubs_subs, bdata::make(core_types_single), core_type) {
+    SetupTest<helics::ValueFederate>(core_type, 1);
+    auto vFed1 = GetFederateAs<helics::ValueFederate>(0);
+
+    auto pubid1 = vFed1->registerPublicationIndexed<double>("pub1", 0);
+    pubid1.setInfo(std::string("pub_test1"));
+
+    auto sub1 = vFed1->registerSubscriptionIndexed("pub1", 0);
+    auto sub2 = vFed1->registerSubscriptionIndexed("pub1", 1);
+    auto sub3 = vFed1->registerSubscriptionIndexed("pub1", 2);
+
+    sub1.setInfo(std::string("sub_test1"));
+    sub2.setInfo(std::string("sub_test2"));
+    sub3.setInfo(std::string("sub_test3"));
+
+    vFed1->enterExecutingMode();
+
+    auto info1 = vFed1->getInfo(pubid1.getHandle());
+    auto info2 = vFed1->getInfo(sub1.getHandle());
+    auto info3 = vFed1->getInfo(sub2.getHandle());
+    auto info4 = vFed1->getInfo(sub3.getHandle());
```

Comment:
This should probably call the getInfo directly on the object vs the fed call

Reply:
I'll add direct requests to the subs themselves, but I think I'll keep the versions that request against the federate, since it gives us better coverage of possible use cases.

Reply:
that sounds fine we should test both ways